File: Rainmaker2/src/MicroServices/ByteWebConnector/ByteWebConnector.API/Macros/Residence.js
if (ev.TableAndFieldName == "Residence.City" || ev.TableAndFieldName == "Residence.State" || ev.TableAndFieldName == "Residence.Country"
    || ev.TableAndFieldName == "Residence.Street" || ev.TableAndFieldName == "Residence.Zip" || ev.TableAndFieldName == "Residence.Current" || ev.TableAndFieldName == "Residence.LivingStatus"
    || ev.TableAndFieldName == "Residence.NoMonths" || ev.TableAndFieldName == "Residence.NoYears" || ev.TableAndFieldName == "Residence.MonthlyRent") {

    var city = ev.Table.GetFieldValue("City");
    var country = ev.Table.GetFieldValue("Country");
    var state = ev.Table.GetFieldValue("State");
    var street = ev.Table.GetFieldValue("Street");
    var zip = ev.Table.GetFieldValue("Zip");
    var current = ev.Table.GetFieldValue("Current");
    var livingStatus = ev.Table.GetField("LivingStatus");
    var noMonths = ev.Table.GetFieldValue("NoMonths");
    var noYears = ev.Table.GetFieldValue("NoYears");
    var monthlyRent = ev.Table.GetFieldValue("MonthlyRent");
    var borId = ev.Table.GetFieldValue("BorrowerID");
    var fileDataId = ev.Table.GetFieldValue("FileDataID");
    var useProxy = true;



    var dataRaw = " --data-raw \"{ \"\"city\"\":\"\"{{city}}\"\" ,\"\"country\"\":\"\"{{country}}\"\" , \"\"state\"\":\"\"{{state}}\"\" , \"\"street\"\":\"\"{{street}}\"\" , \"\"zip\"\":\"\"{{zip}}\"\" , \"\"current\"\":\"{{current}}\", \"\"livingStatus\"\":\"{{livingStatus}}\", \"\"noMonths\"\":\"\"{{noMonths}}\"\", \"\"noYears\"\":\"\"{{noYears}}\"\", \"\"monthlyRent\"\":\"\"{{monthlyRent}}\"\", \"\"borrowerID\"\":\"{{borId}}\" , \"\"fileDataId\"\":\"{{fileDataId}}\"  }\" ";

    var arguments = "{{proxy}} --location --request POST \"http://localhost:52537/api/Values/residence\" --header \"Content-Type: application/json\" --header \"Accept: application/json\" {{dataRaw}}";

    if (useProxy) {
        arguments = arguments.replace("{{proxy}}", "--proxy 127.0.0.1:8888");
    }
    dataRaw = dataRaw.replace("{{city}}", city);
    dataRaw = dataRaw.replace("{{country}}", country);
    dataRaw = dataRaw.replace("{{state}}", state);
    dataRaw = dataRaw.replace("{{street}}", street);
    dataRaw = dataRaw.replace("{{zip}}", zip);
    dataRaw = dataRaw.replace("{{current}}", current);
    dataRaw = dataRaw.replace("{{livingStatus}}", livingStatus);
    dataRaw = dataRaw.replace("{{noMonths}}", noMonths);
    dataRaw = dataRaw.replace("{{noYears}}", noYears);
    dataRaw = dataRaw.replace("{{monthlyRent}}", monthlyRent);
    dataRaw = dataRaw.replace("{{borId}}", borId);
    dataRaw = dataRaw.replace("{{fileDataId}}", fileDataId);

    arguments = arguments.replace("{{dataRaw}}", dataRaw);
    los.Application.ShowMessageBox("arguments " + arguments);
    System.Diagnostics.Process.Start("curl.exe", arguments);
};

File: Rainmaker2/src/MicroServices/LosIntegration/LosIntegration.API/Models/Residence.cs
namespace LosIntegration.API.Models
{
    public class Residence
    {
        public int AppNo { get; set; }
        public int ResidenceId { get; set; }
        public int? BorrowerId { get; set; }
        public int DisplayOrder { get; set; }
        public bool Current { get; set; }
        public string FullAddress { get; set; }
        public string CityStateZip { get; set; }
        public string Street { get; set; }
        public string City { get; set; }
        public string State { get; set; }
        public string Zip { get; set; }
        public string LivingStatus { get; set; }
        public int? NoYears { get; set; }
        public int? NoMonths { get; set; }
        public string LLName { get; set; }
        public string LLAttn { get; set; }
        public string LLFullAddress { get; set; }
        public string LLCityStateZip { get; set; }
        public string LLStreet { get; set; }
        public string LLCity { get; set; }
        public string LLState { get; set; }
        public string LLZip { get; set; }
        public string LLPhone { get; set; }
        public string Notes { get; set; }
        public string LLFax { get; set; }
        public string Country { get; set; }
        public int? MonthlyRent { get; set; }
        public int StreetContainsUnitNumberOv { get; set; }
        public long FileDataId { get; set; }


        public object GetRainmakerBorrowerResidence() 
        {
            throw new System.NotImplementedException();
        }
    }
}
